src/topicButton.js

```diff
@@ -109,4 +131,63 @@ var TopicButton = GObject.registerClass({
         this._attentionSign.revealChild = value;
         this.notify('needs-attention');
     }
+
+    get revealed() {
+        return this._revealed;
+    }
+
+    set revealed(value) {
+        if ('_revealed' in this && this._revealed === value)
+            return;
+
+        this._revealed = value;
+        this.notify('revealed');
+    }
+
+    get sensitive() {
+        return this._sensitive;
+    }
+
+    set sensitive(value) {
+        if ('_sensitive' in this && this._sensitive === value)
+            return;
+
+        this._sensitive = value;
+        this.notify('sensitive');
+
+        if (!this._sensitive)
+            this.get_style_context().add_class('insensitive');
+        else if (this._sensitive)
+            this.get_style_context().remove_class('insensitive');
+    }
+
+    dbusRegister(appId) {
```

Comment:
Great, this is a much better way to get the app ID into this function than I thought there was going to be.